client/app/scripts/superdesk/editor/editor.js

```diff
@@ -199,11 +203,125 @@ function FindReplaceCommand(rootNode) {
     }
 }
 
+function SpellcheckService($q) {
+
+    var selection = document.getSelection();
+
+    this.check = function check(text) {
+        var words = _.uniq(_.filter(text.split(/[\W0-9]+/)));
+        var errors = _.sample(words, Math.min(5, words.length - 1));
+        return errors;
+    };
+
+    function cursorReset(node) {
+        selection.collapse(node, 0);
+    }
+
+    function cursorMove(times) {
+        for (var i = 0; i < times; i++) {
+            selection.modify('move', 'right', 'character');
```

Comment:
`selection.modify()` is [not standard](https://developer.mozilla.org/en-US/docs/Web/API/Selection/modify). We must be careful and test this across different browsers.